Carting/src/WebApi/Application/Carts/Commands/RemoveItemFromCartCommand/RemoveItemFromCartCommand.cs

```diff
@@ -0,0 +1,41 @@
+﻿using Carting.WebApi.Application.Common.Exceptions;
+using Carting.WebApi.Application.Common.Interfaces;
+using Carting.WebApi.Domain.Entities;
+using MediatR;
+
+namespace Carting.WebApi.Application.Carts.Commands.RemoveItemFromCartCommand;
+
+public record RemoveItemFromCartCommand : IRequest<string>
+{
+    public string CartId { get; init; }
+    public int Id { get; init; }
+}
+
+public class RemoveItemFromCartCommandHandler : IRequestHandler<RemoveItemFromCartCommand, string>
+{
+    private readonly ICartingDbContext _context;
+
+    public RemoveItemFromCartCommandHandler(ICartingDbContext context)
+    {
+        _context = context;
+    }
+
+
+    Task<string> IRequestHandler<RemoveItemFromCartCommand, string>.Handle(RemoveItemFromCartCommand request, CancellationToken cancellationToken)
+    {
+        var cart = _context.Get<Cart>(request.CartId);
```

Comment:
Unfortunately, LiteDB doesn't support async methods. But I understand that DB access is typically done via async methods and this way the implementation (LiteDB) has an affect on other parts of the system (Application layer). Do you think it would be feasable to make the methods of `ICartingDbContext` asynchronous and bear with the little inefficiency it causes?

Reply:
If LiteDb doesn't support this so, no, as this is a demo application, unless you plan to expand this application beyond the current tasks. In general production environment prefer async methods